tests/naive.py

```diff
@@ -267,71 +267,107 @@ def stump(T_A, m, T_B=None, exclusion_zone=None, row_wise=False, k=1):
     return result
 
 
-def aamp(T_A, m, T_B=None, exclusion_zone=None, p=2.0):
+def aamp(T_A, m, T_B=None, exclusion_zone=None, p=2.0, row_wise=False, k=1):
+    """
+    Traverse distance matrix diagonally and update the top-k matrix profile and
+    matrix profile indices if the parameter `row_wise` is set to `False`. If the
+    parameter `row_wise` is set to `True`, it is a row-wise traversal.
+    """
     T_A = np.asarray(T_A)
     T_A = T_A.copy()
 
     if T_B is None:
-        T_B = T_A.copy()
         ignore_trivial = True
+        T_B = T_A.copy()
     else:
+        ignore_trivial = False
         T_B = np.asarray(T_B)
         T_B = T_B.copy()
-        ignore_trivial = False
 
     T_A[np.isinf(T_A)] = np.nan
     T_B[np.isinf(T_B)] = np.nan
 
     rolling_T_A = core.rolling_window(T_A, m)
     rolling_T_B = core.rolling_window(T_B, m)
 
+    distance_matrix = cdist(rolling_T_A, rolling_T_B, metric="minkowski", p=p)
+
     n_A = T_A.shape[0]
     n_B = T_B.shape[0]
     l = n_A - m + 1
     if exclusion_zone is None:
         exclusion_zone = int(np.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
 
-    distance_matrix = cdist(rolling_T_A, rolling_T_B, metric="minkowski", p=p)
+    P = np.full((l, k + 2), np.inf, dtype=np.float64)
+    I = np.full((l, k + 2), -1, dtype=np.int64)  # two more columns are to store
+    # ... left and right top-1 matrix profile indices
 
-    if ignore_trivial:
-        diags = np.arange(exclusion_zone + 1, n_A - m + 1)
-    else:
-        diags = np.arange(-(n_A - m + 1) + 1, n_B - m + 1)
+    if row_wise:
```

Comment:
To be consistent with `naive.stump`, I added param `row_wise` to `naive.aamp`. Then, I used `row_wise==True` for testing `gpu_aamp`